common.tf

```diff
@@ -168,3 +170,59 @@ module "files_table" {
     }
   ]
 }
+
+data "aws_ssm_parameter" "slack_token" {
+  name            = "/mgmt/slack/token"
+  with_decryption = true
+}
+
+data "aws_ssm_parameter" "dr2_notifications_slack_channel" {
+  name = "/mgmt/slack/notifications/channel"
+}
+
+module "eventbridge_alarm_notifications_destination" {
+  source                     = "git::https://github.com/nationalarchives/da-terraform-modules//eventbridge_api_destination"
+  authorisation_header_value = "Bearer ${data.aws_ssm_parameter.slack_token.value}"
+  name                       = "${local.environment}-eventbridge-slack-destination"
+}
+
+module "cloudwatch_alarm_event_bridge_rule" {
+  for_each = toset(["OK", "ALARM"])
+  source   = "git::https://github.com/nationalarchives/da-terraform-modules//eventbridge_api_destination_rule"
+  event_pattern = templatefile("${path.module}/templates/eventbridge/cloudwatch_alarm_event_pattern.json.tpl", {
+    cloudwatch_alarms = jsonencode(flatten([
+      module.download_files_sqs.dlq_cloudwatch_alarm_arn,
+      module.ingest_parsed_court_document_event_handler_sqs.dlq_cloudwatch_alarm_arn,
+      module.preservica_config_queue.dlq_cloudwatch_alarm_arn
+    ])),
+    state_value = each.value
+  })
+  name                = "${local.environment}-eventbridge-alarm-state-change-${lower(each.value)}"
+  api_destination_arn = module.eventbridge_alarm_notifications_destination.api_destination_arn
+  input_transformer = {
+    input_paths = {
+      "alarmName"    = "$.detail.alarmName",
+      "currentValue" = "$.detail.state.value"
+    }
+    input_template = templatefile("${path.module}/templates/eventbridge/slack_message_input_template.json.tpl", {
+      channel_id = data.aws_ssm_parameter.dr2_notifications_slack_channel.value
+      message    = ":${each.value == "OK" ? "green-tick" : "alert-noflash-slow"}: Cloudwatch alarm <alarmName> has entered state <currentValue>"
+    })
+  }
+}
+
+module "dev_slack_message_eventbridge_rule" {
+  source              = "git::https://github.com/nationalarchives/da-terraform-modules//eventbridge_api_destination_rule"
+  api_destination_arn = module.eventbridge_alarm_notifications_destination.api_destination_arn
+  event_pattern       = templatefile("${path.module}/templates/eventbridge/custom_detail_type_event_pattern.json.tpl", { detail_type = "DR2DevMessage" })
+  name                = "${local.environment}-eventbridge-dev-slack-message"
+  input_transformer = {
+    input_paths = {
+      "message" = "$.detail.message"
```

Comment:
Should we rename either this "message" or the one on line 225, to "slackMessage"?

Reply:
We can change this one and the one on line 255. I've changed them both.